tests/cursorid/cursorid-set_state-001.phpt

```diff
@@ -0,0 +1,21 @@
+--TEST--
+MongoDB\Driver\CursorId::__set_state()
+--FILE--
+<?php
+
+$cursorId = unserialize('C:23:"MongoDB\Driver\CursorId":42:{a:1:{s:2:"id";s:19:"7250031947823432848";}}');
+
+var_export(MongoDB\Driver\CursorId::__set_state([
+    'id' => "$cursorId",
```

Comment:
Out of curiosity, how/why did you decide to use `"$cursorId"` here?

This happens to work because double quoted strings will convert the `$cursorId` variable to a string per PHP's [string parsing semantics](https://www.php.net/manual/en/language.types.string.php#language.types.string.parsing). We typically use single-quoted strings in the driver/library whenever possible to avoid such behavior (and avoid any chance of it happening inadvertently).

If you look at the [`__toString` magic method](https://www.php.net/manual/en/language.oop5.magic.php#object.tostring) on the CursorId class, you'll see that it returns the internal 64-bit integer as a string value.

There's really no need for this test to exercise `CursorId::__toString` and unserialization. You could simply this by using `'7250031947823432848'` here directly.

----

Having said all that, it looks like we might not have another test that covers `CursorId::__toString` so it'd be worth adding one (e.g. `cursorid-tostring-001.phpt`) that does something like:

```php
$cursorId = unserialize('C:23:"MongoDB\Driver\CursorId":42:{a:1:{s:2:"id";s:19:"7250031947823432848";}}');
var_dump((string) $cursorId);
```

Reply:
Ah ok, I didn't have any clear reasoning behind using `"$cursorId"` - just figured that it would work as a string. I will add the `__toString` test!